packages/core/src/index.ts

```diff
@@ -288,6 +288,31 @@ class Hopr extends EventEmitter {
     }
   }
 
+  /**
+   * If error provided is considered an out of funds error
+   * - it will emit that the node is out of funds
+   * - it will set channel strategy to passive
+   * @param error error thrown by an ethereum transaction
+   */
+  private async isOutOfFunds(error: any): Promise<void> {
+    const isOutOfFunds = isErrorOutOfFunds(error)
+    if (!isOutOfFunds) return
+
+    const address = (await this.getEthereumAddress()).toHex()
+
+    if (isOutOfFunds === 'NATIVE') {
+      log('unfunded node', address)
+      this.emit('hopr:warning:unfundedNative', address)
+      await this.getNativeBalance()
```

Comment:
How do we use the result? I guess it triggers some side-effects?

Reply:
it gets picked up by `hoprd` and displayed to the user

Reply:
might be nice to have a comment that quickly describe the side-effect

Reply:
not the responsibility of core - hoprd is a child application. As long as the event is known, this is sufficient.